[FIX] wrong scan line

eLBati opened the pull request:
Without this change, invoice with total amount of 4502.47 CHF produces the line
00014502475>...

With this change, it produces the line
0100004502471>...

eLBati committed:
[FIX] wrong scan line

Without this change, invoice with total amount of 4502.47 CHF produces the line
00014502475>...

With this change, it produces the line
0100004502471>...

nbessi commented:
After reading https://www.postfinance.ch/binp/postfinance/public/dam.Up1Ct9TDDT8MFk4AB6m00phIHrTLJlvd0XyZjBIXy3M.spool/content/dam/pf/de/doc/consult/manual/dlserv/inpayslip_isr_man_fr.pdf 

fix seems correct let see if test are green.

nbessi commented on l10n_ch_payment_slip/payment_slip.py:
```diff
@@ -202,8 +202,8 @@ def _compute_scan_line_list(self):
         if not self._can_generate(self.move_line_id):
             return []
-        amount = '01%.2f' % self.amount_total
-        justified_amount = amount.replace('.', '').rjust(10, '0')
```
We have to round the amount_total to 2 digit to ensure bvr is on 2 digit it precision is changed.

nbessi commented:
:+1:

nbessi commented:
@eLBati it would be really nice if you can add a test that check scan line see #93

eLBati committed:
[FIX] wrong use of unicode():

>>> amount=64.80
>>> '01%s' % unicode(amount).replace('.', '').rjust(10, '0')
u'010000000648'

it should be:

>>> '01%s' % ('%.2f' % amount).replace('.', '').rjust(10, '0')
'010000006480'

guewen commented:
:+1:
